### ISSUE TITLE
Does combineLatest unsubscribe from one stream if another ends with error?

### ISSUE THREAD

**Issue by leventov** [ReactiveX/RxJava#6818]
Javadoc and ReactiveX docs for `combineLatest` and a related `zip` operator don't elaborate this. It's hard to follow the implementation code for those unfamiliar with the internals of the lib to find an answer for themselves.

**Comment by akarnokd** [comment]
[Yes](https://github.com/ReactiveX/RxJava/blob/3.x/src/main/java/io/reactivex/rxjava3/internal/operators/observable/ObservableCombineLatest.java#L251).

**Comment by leventov** [comment]
Thanks for a quick response.

Maybe a note could be added to javadocs about it?

**Comment by leventov** [comment]
Could you please also kindly respond about rxJava 1.x? [That's the code I was actually looking at](https://github.com/ReactiveX/RxJava/blob/1.x/src/main/java/rx/internal/operators/OnSubscribeCombineLatest.java), and it's harder to see if the unsubscription happens there. (We use 1.x in prod.)

**Comment by akarnokd** [comment]
1.x is no longer supported but [here it is](https://github.com/ReactiveX/RxJava/blob/1.x/src/main/java/rx/internal/operators/OnSubscribeCombineLatest.java#L175).

> Maybe a note could be added to javadocs about it?

Perhaps.

**Comment by leventov** [comment]
I can do it. To what other combining operators this also apply? Or if to most, to what it does not apply?

**Comment by akarnokd** [comment]
Flows are not left hanging in general; they either terminate on their own or get disposed when they are no longer needed. Usually we don't document this as there is already a rule in the Reactive Streams specification: https://github.com/reactive-streams/reactive-streams-jvm#2.6
